packages/core/plugin/src/withSentryAndroidGradlePlugin.ts

```diff
@@ -0,0 +1,96 @@
+import { withAppBuildGradle, withProjectBuildGradle } from '@expo/config-plugins';
+
+export interface SentryAndroidGradlePluginOptions {
+  sentryAndroidGradlePluginVersion?: string;
+  includeProguardMapping?: boolean;
+  dexguardEnabled?: boolean;
+  autoUploadNativeSymbols?: boolean;
+  autoUploadProguardMapping?: boolean;
+  uploadNativeSymbols?: boolean;
+  includeNativeSources?: boolean;
+  includeSourceContext?: boolean;
+  autoInstallationEnabled?: boolean;
+}
+
+/**
+ * Adds the Sentry Android Gradle Plugin to the project.
+ * https://docs.sentry.io/platforms/react-native/manual-setup/manual-setup/#enable-sentry-agp
+ */
+export function withSentryAndroidGradlePlugin(config: any, options: SentryAndroidGradlePluginOptions = {}): any {
+  const version = options.sentryAndroidGradlePluginVersion;
+  const includeProguardMapping = options.includeProguardMapping ?? true;
+  const dexguardEnabled = options.dexguardEnabled ?? false;
+  const autoUploadProguardMapping = options.autoUploadProguardMapping ?? true;
+  const uploadNativeSymbols = options.uploadNativeSymbols ?? true;
+  const autoUploadNativeSymbols = options.autoUploadNativeSymbols ?? true;
+  const includeNativeSources = options.includeNativeSources ?? true;
+  const includeSourceContext = options.includeSourceContext ?? false;
+  const autoInstallationEnabled = options.autoInstallationEnabled ?? false;
+
+  // Modify android/build.gradle
+  const withSentryProjectBuildGradle = (config: any): any => {
+    return withProjectBuildGradle(config, (projectBuildGradle: any) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (!projectBuildGradle.modResults || !projectBuildGradle.modResults.contents) {
+        throw new Error('android/build.gradle content is missing or undefined.');
+      }
+
+      const dependency = `classpath("io.sentry:sentry-android-gradle-plugin:${version}")`;
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (!projectBuildGradle.modResults.contents.includes(dependency)) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        projectBuildGradle.modResults.contents = projectBuildGradle.modResults.contents.replace(
+          /dependencies\s*{/,
+          `dependencies {\n        ${dependency}`,
+        );
```

Comment:
Let's also add a warning if we could not add the dependency.

Reply:
Added with cf55ce46f05293317a397cd5b138dfac423d9313 and f04a2c74709f125bb27253ad4aa3bc7c6def35bc